### PR TITLE
Netcore 3.1 Support

### PR DESCRIPTION
**Pull request by AffectedArc07**
Done awfully but atleast its done. This will allow me to take entire RAM dumps of the process which can ease in figuring out why the hell this happens
image

### CODE REVIEW

File: IconDiffBot/Core/Application.cs [hunk 1]
```diff
@@ -57,12 +59,15 @@ public void ConfigureServices(IServiceCollection services)
...
 			});
 
+			services.AddHangfireServer();
+
 			services.Configure<IISOptions>((options) => options.ForwardClientCertificate = false);
-			services.AddMvc();
 			services.AddOptions();
 			services.AddLocalization();
+			services.AddControllersWithViews();
+			services.AddRazorPages();
```
Comment (Cyberboss) [inline IconDiffBot/Core/Application.cs:68]: Why'd you change AddMvc with these two?

File: IconDiffBot/Core/Application.cs [hunk 2]
```diff
@@ -57,12 +59,15 @@ public void ConfigureServices(IServiceCollection services)
...
 
-			services.AddDbContext<DatabaseContext>();
+			services.AddDbContext<DatabaseContext>(ServiceLifetime.Transient);
```
Comment (Cyberboss) [inline IconDiffBot/Core/Application.cs:70]: Weird, that should've been the default option (If it wasn't that alone is a potentially big leak)

File: IconDiffBot/Core/Application.cs [hunk 3]
```diff
@@ -126,7 +131,10 @@ public void Configure(IApplicationBuilder applicationBuilder, IHostingEnvironmen
 					Authorization = new List<IDashboardAuthorizationFilter> { }
 				});
 
-			applicationBuilder.UseMvc();
+			applicationBuilder.UseRouting();
+			applicationBuilder.UseEndpoints(endpoints => {
+				endpoints.MapControllers();
+			});
```
Comment (Cyberboss) [inline IconDiffBot/Core/Application.cs:137]: I don't know if you tested this, but I was under the impressing that you needed to disable endpoint routing.

If it works it works though

File: IconDiffBot/Core/Application.cs (hunk 1 above)
Comment (AffectedArc07) [reply IconDiffBot/Core/Application.cs:68]: I needed to change it to these two for it to work, and the migration said guide said so

File: IconDiffBot/Core/Application.cs (hunk 2 above)
Comment (AffectedArc07) [reply IconDiffBot/Core/Application.cs:70]: I had to add this to make the DBContext not complain about being threaded.

File: IconDiffBot/Core/Application.cs (hunk 3 above)
Comment (AffectedArc07) [reply IconDiffBot/Core/Application.cs:137]: This is the only way I *could* make it work

File: IconDiffBot/Core/Application.cs (hunk 2 above)
Comment (bobbah) [reply IconDiffBot/Core/Application.cs:70]: i would be surprised if it wasn't the default, you'd run into issues with the db connection already being open/in-use as dbcontext are not threadsafe

**Review by bobbah — COMMENTED** [review]
[update this line](https://github.com/tgstation/IconDiffBot/pull/61/files#diff-d0055522b17402cfdf734259600bd008R21)

it won't point to the right dir when you build with 3.1

**Comment by AffectedArc07** [comment]
Docker failing is not an issue. We dont run docker in prod and it can be fixed at a later date.

It also 404s when trying to look at the logs, so I declare it shit CI

**Pull request merged**